shared-libs/transitions/src/transitions/mark_for_outbound.js

```diff
@@ -1,24 +1,58 @@
+/**
+ * Attempts to convert and send an outbound request, queuing it for later if needed.
+ *
+ * For historic reasons this is called *mark* for outbound (as opposed to send and then if that
+ * fails mark for outbound, but transition names are part of our API so we can't change it easily.
+ *
+ */
 const vm = require('vm');
 
 const config = require('../config');
 const db = require('../db');
 const logger = require('../lib/logger');
+const outbound = require('@medic/outbound')(logger);
 
 const CONFIGURED_PUSHES = 'outbound';
 
 const relevantTo = doc => {
   const pushes = config.get(CONFIGURED_PUSHES) || {};
 
-  return Object.keys(pushes).filter(key => {
-    const conf = pushes[key];
-    return conf.relevant_to && vm.runInNewContext(conf.relevant_to, {doc});
-  });
+  return Object.keys(pushes)
+    .filter(key => {
+      const conf = pushes[key];
+      return conf.relevant_to && vm.runInNewContext(conf.relevant_to, {doc});
+    })
+    .map(key => [pushes[key], key]);
 };
 
 const markForOutbound = (change) => {
-  const toQueue = relevantTo(change.doc);
+  // We're working out the relevant tasks to perform here and not in the exported filter function,
+  // because there is no way to communicate between the filter and onMatch functions, and so because
+  // we *have* to do it here nothing of value is added to also perform it in filter
+  const relevantConfigs = relevantTo(change.doc);
+
+  let p = Promise.resolve();
+
+  const failedKeys = [];
+
+  for (const [config, configKey] of relevantConfigs) {
+    p = p.then(() => outbound.send(config, configKey, change.doc, change.info))
+      .then(() => {
+        // Successful, outbound.send wrote to the infodoc
+
+        return db.sentinel.put(change.info);
+      })
+      .catch(() => {
+        // First realtime attempt failed, queue for later
+        failedKeys.push(configKey);
+      });
+  }
+
+  return p.then(() => {
```

Comment:
Nit: I think this would be nicer with async/await as you wouldn't need to chain promises, just loop and await.

Reply:
To do that I'd have to bring that eslint rule I just moved into the tests back. I think we'll skip it for now until we are more comfortable moving the eslint rules forward